tests/sysprep_test.go

```diff
@@ -0,0 +1,361 @@
+    <RestartEnabled>true</RestartEnabled>
+    </component>
+    <component name="Microsoft-Windows-ehome-reg-inf" processorArchitecture="amd64" publicKeyToken="31bf3856ad364e35" language="neutral" versionScope="NonSxS" xmlns:wcm="http://schemas.microsoft.com/WMIConfig/2002/State" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <RestartEnabled>true</RestartEnabled>
+    </component>
+    </settings>
+    </unattend>
+    `
+)
+
+func insertProductKeyToAnswerFileTemplate(answerFileTemplate string) string {
+	productKeyFilePath := os.Getenv("KUBEVIRT_WINDOWS_PRODUCT_KEY_PATH")
+	keyFromFile, err := ioutil.ReadFile(productKeyFilePath)
+	Expect(err).ToNot(HaveOccurred())
+	productKey := strings.TrimSpace(string(keyFromFile))
+	return fmt.Sprintf(answerFileTemplate, productKey)
+}
+
+var getWindowsSysprepVMISpec = func() v1.VirtualMachineInstanceSpec {
+	gracePeriod := int64(0)
+	spinlocks := uint32(8191)
+	firmware := types.UID(windowsFirmware)
+	_false := false
+	return v1.VirtualMachineInstanceSpec{
+		TerminationGracePeriodSeconds: &gracePeriod,
+		Domain: v1.DomainSpec{
+			CPU: &v1.CPU{Cores: 2},
+			Features: &v1.Features{
+				ACPI: v1.FeatureState{},
+				APIC: &v1.FeatureAPIC{},
+				Hyperv: &v1.FeatureHyperv{
+					Relaxed:   &v1.FeatureState{},
+					VAPIC:     &v1.FeatureState{},
+					Spinlocks: &v1.FeatureSpinlocks{Retries: &spinlocks},
+				},
+			},
+			Clock: &v1.Clock{
+				ClockOffset: v1.ClockOffset{UTC: &v1.ClockOffsetUTC{}},
+				Timer: &v1.Timer{
+					HPET:   &v1.HPETTimer{Enabled: &_false},
+					PIT:    &v1.PITTimer{TickPolicy: v1.PITTickPolicyDelay},
+					RTC:    &v1.RTCTimer{TickPolicy: v1.RTCTickPolicyCatchup},
+					Hyperv: &v1.HypervTimer{},
+				},
+			},
+			Firmware: &v1.Firmware{UUID: firmware},
+			Resources: v1.ResourceRequirements{
+				Requests: k8sv1.ResourceList{
+					k8sv1.ResourceMemory: resource.MustParse("2048Mi"),
+				},
+			},
+			Devices: v1.Devices{
+				Disks: []v1.Disk{
+					{
+						Name:       windowsSealedDisk,
+						DiskDevice: v1.DiskDevice{Disk: &v1.DiskTarget{Bus: "sata"}},
+					},
+					{
+						Name:       "sysprep",
+						DiskDevice: v1.DiskDevice{CDRom: &v1.CDRomTarget{Bus: "sata"}},
+					},
+				},
+			},
+		},
+		Volumes: []v1.Volume{
+			{
+				Name: windowsSealedDisk,
+				VolumeSource: v1.VolumeSource{
+					Ephemeral: &v1.EphemeralVolumeSource{
+						PersistentVolumeClaim: &k8sv1.PersistentVolumeClaimVolumeSource{
+							ClaimName: tests.DiskWindowsSysprep,
+						},
+					},
+				},
+			},
+			{
+				Name: "sysprep",
+				VolumeSource: v1.VolumeSource{
+					Sysprep: &v1.SysprepSource{
+						ConfigMap: &k8sv1.LocalObjectReference{
+							Name: "sysprepautounattend",
+						},
+					},
+				},
+			},
+		},
+	}
+
+}
+
+const (
+	windowsSysprepedVMIUser     = "Admin"
+	windowsSysprepedVMIPassword = "Gauranga"
+)
+
+var _ = Describe("[Serial][Sysprep]Windows VirtualMachineInstance", func() {
```

Comment:
@borod108 Recently we've adopted a rule of thumb to assign a SIG label to new and existing functional tests in order to classify them according to our CI lanes. I believe this test belongs to `sig-compute`. Can you please add `[sig-comute]` label in the Describe message?

Reply:
Sure, done.

Reply:
@enp0s3 question: this test should only run in a specific lane, with a specific volume mounted: https://github.com/kubevirt/project-infra/pull/1280
So now that it is marked as sig-compute it will run and fail every time the focus is sig-compute.... what should be done about that?

Reply:
I think we should not mark this one as `[sig-compute]`, same as we do with the other windows test. Having `Windows` in the description (so that we can filter it out for regular lanes) and its own tag (so that we can execute it in its own lane) should be enough.

Reply:
@enp0s3 is this ok?
Also - I think the tests that are failing have nothing to do with this PR...

Reply:
> I think we should not mark this one as `[sig-compute]`, same as we do with the other windows test.

Sorry, this is wrong, the other Windows test is marked as `[sig-compute]`, we should do the same here as @enp0s3 suggests.

> So now that it is marked as sig-compute it will run and fail every time the focus is sig-compute.... what should be done about that?

@borod108 Are you sure it was triggered as part of the sig-compute tests? This doesn't happen with the existing Windows test, it should be the same.

Reply:
@fgimenez regarding this point - if I mark this as sig-compute, won't it run on every push?

Reply:
@enp0s3 so on this point - I see that I have to add a sig label, but I do not want this test to run every push, what to do? (Sig compute runs every time right?)
Apart from this point, can is everything else ok with you?

Reply:
@borod108 Hi, I believe the sig label you need to add is `sig-compute`. If you want this test to be skipped under some circumstances you can use Ginkgo `Skip(..)` function. If you want this test to be run only in specific lanes you should take care of it either in `test/automation` and also in project-infra repo.

Reply:
@enp0s3 thank you! fixed.